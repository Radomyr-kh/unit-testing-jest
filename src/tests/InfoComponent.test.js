import getGitHubUser from '../services/DataService/index';
import Info from '../components/Info/index';
import renderer from 'react-test-renderer';
import mockAxios from 'axios';

describe('Tests for Info Component', () => {
  let wrapper;
  let userName;

  beforeEach(() => {
    userName = 'some-name';
    wrapper = renderer.create(<Info user={userName} />);
  });

  it('New InfoComponent should be an instance of InfoComponent', () => {
    expect(new Info()).toBeInstanceOf(Info);
  });

  it('renders correctly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders h3 = "GitHub User Info"', () => {
    const instance = wrapper.root;
    const element = instance.findByType('h3');
    expect(element.props.children).toEqual('GitHub User Info');
  });

  it('component contains user prop', () => {
    const tree = wrapper.toTree();
    expect(tree.props.user).toEqual(userName);
  });

  it('error is defined', () => {
    const instance = wrapper.getInstance();
    expect(instance.state.error).toBeDefined();
  });

  // mock axios
  it('InfoComponent renders all items from mock data correctly', async () => {
    mockAxios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: {
          login: 'someLogin',
          id: '12345',
          followers: '43',
        },
      })
    );
    const res = await getGitHubUser();

    const list = {
      login: 'someLogin',
      id: '12345',
      followers: '43',
    };
    expect(res.data).toEqual(list);
  });
});
