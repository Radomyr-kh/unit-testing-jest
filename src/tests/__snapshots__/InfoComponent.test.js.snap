// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Info Component renders correctly 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul>
    <li>
      error
      : 
      request error
    </li>
  </ul>
</div>
`;
